tests/pytorch/test_onnx_export.py

```diff
@@ -801,6 +801,27 @@ def test_export_layernorm_mlp(
         else:
             validate_result(fname, inp, model, atol=2e-2, is_fp8=use_fp8)
 
+def export_and_validate_flash_attn_models(
+        model: torch.nn.Module,
+        inp: Union[Tuple[torch.Tensor], torch.Tensor],
+        fname: str,
+        atol: float,
+        input_names: list=["input"],
+        export_fp8: bool=False,
+        validate_fp8: bool=False):
+    '''
+    Disabling usage of fused attention (flash_attention) as writing ONNX symbolic functions requires changes in the flash-attention library.
+    It is legal to do this because TE should behave the same when flash-attention is used and when it is not used.
+    We make sure to compare the results of TE with flash-attention to ONNX w/o flash attention, to ensure that this is legal.
+    '''
+    with te.onnx_export(True):
```

Comment:
I think you can remove `export_and_validate_flash_attn_models` and just have the default export use the context manager like here.
TE modules that have special behavior on export will trigger the correct module behavior, and the rest of the modules will just ignore this export mode.